test/src/test/java/io/pravega/schemaregistry/integrationtest/TestEndToEndWithAuth.java

```diff
@@ -0,0 +1,111 @@
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.URI;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Base64;
+import java.util.concurrent.Executors;
+
+@Slf4j
+public class TestEndToEndWithAuth extends TestInMemoryEndToEnd {
+    private final static String SYSTEM_ADMIN = "admin";
+    private final static String PASSWORD = "1111_aaaa";
+
+    protected int port;
+    private RestServer restServer;
+    private File authFile;
+
+    @Before
+    @Override
+    public void setUp() {
+        executor = Executors.newScheduledThreadPool(Config.THREAD_POOL_SIZE);
+
+        port = TestUtils.getAvailableListenPort();
+        authFile = createAuthFile();
+        ServiceConfig serviceConfig = ServiceConfig.builder().port(port)
+                                                   .authEnabled(true)
+                                                   .userPasswordFilePath(authFile.getAbsolutePath())
+                                                   .build();
+        SchemaStore store = getStore();
+
+        SchemaRegistryService service = new SchemaRegistryService(store, executor);
+
+        restServer = new RestServer(service, serviceConfig);
+        restServer.startAsync();
+        restServer.awaitRunning();
+    }
+
+    @After
+    public void tearDown() {
+        restServer.stopAsync();
+        restServer.awaitTerminated();
+        executor.shutdownNow();
+        authFile.delete();
+    }
+
+    @Override
+    public SchemaRegistryClient getClient() {
+        return SchemaRegistryClientFactory.withDefaultNamespace(
+                SchemaRegistryClientConfig.builder().schemaRegistryUri(URI.create("http://localhost:" + port))
+                                          .authentication(new CredentialProvider.DefaultCredentialProvider("Basic", 
+                                                  Base64.getEncoder().encodeToString((SYSTEM_ADMIN + ":" + PASSWORD).getBytes(Charsets.UTF_8))))
+                                          .build());
+    }
+    
+    private File createAuthFile() {
+        try {
+            File authFile = File.createTempFile("auth_file", ".txt");
+            StrongPasswordProcessor passwordEncryptor = StrongPasswordProcessor.builder().build();
+
+            try (FileWriter writer = new FileWriter(authFile.getAbsolutePath())) {
+                String defaultPassword = passwordEncryptor.encryptPassword(PASSWORD);
+                writer.write(credentialsAndAclAsString(SYSTEM_ADMIN, defaultPassword, "prn::*,READ_UPDATE;"));
+            }
+            return authFile;
+        } catch (IOException | NoSuchAlgorithmException | InvalidKeySpecException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String credentialsAndAclAsString(String username, String password, String acl) {
+        Preconditions.checkArgument(!Strings.isNullOrEmpty(username)
```

Comment:
nit: Since this is a private method, the input validation isn't necessary. More so as this is as a test method.